Grid config: set values from startup and improve layout

chapulina opened the pull request:
# 🎉 New feature

## Summary
<!--Explain changes made, the expected behavior, and provide any other additional
context (e.g., screenshots, gifs) if appropriate.-->

* Make widgets expand to fill the entire plugin
* Increase maximum values
* Make use of `sceneFromFirstRenderEngine` to reduce duplication
* Choose number of decimals according to widget width
* Make it possible to set initial values from XML
* Add example

Note to self: backport to `ign-gazebo3`

## Test it
<!--Explain how reviewers can test this new feature manually.-->

`ign gui -v 4 -c ign-gui/examples/config/grid_config.config`

![grid_config](https://user-images.githubusercontent.com/5751272/144502085-8634ff23-7012-4f73-8e2d-06b522dbd835.gif)

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [x] ~~Updated migration guide (as needed)~~
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

ahcorde commented on src/plugins/grid_config/GridConfig.hh:
```diff
@@ -77,6 +77,25 @@ namespace gui
+    /// \param[in] _hCellCount Horizontal cell count
+    /// \param[in] _vCellCount Vertical cell count
+    /// \param[in] _cellLength Cell length
+    /// \param[in] _x X position
+    /// \param[in] _y Y position
+    /// \param[in] _z Z position
+    /// \param[in] _roll Orientation's roll
+    /// \param[in] _pitch Orientation's pitch
+    /// \param[in] _yaw Orientation's yaw
```
```suggestion
    /// \param[in] _pos
    /// \param[in] _rot
```

ahcorde commented on examples/config/grid_config.config:
```diff
@@ -0,0 +1,36 @@
+  <cell_count>3</cell_count>
+  <vertical_cell_count>2</vertical_cell_count>
+  <cell_length>2</cell_length>
+  <pose>1 2 3  0 0 0</pose>
```
```suggestion
  <pose>1 2 3 0 0 0</pose>
```

chapulina replied:
I've been trying to get in the habit of separating position and rotation with an extra space as suggested on [Specifying pose: Proposal for a better pose](http://sdformat.org/tutorials?tut=better_pose_proposal&cat=pose_semantics_docs&#motivation):

_libsdformat and SDFormat tutorials should encourage additional whitespace, e.g. to separate translation and rotation, use 3 spaces (instead of 1) as a delimiter between values if they fit on one line, or use a newline (possibly with hanging indents) if they do not fit on one line._

jennuine commented on examples/config/grid_config.config:
```diff
@@ -0,0 +1,36 @@
+    <property key="showTitleBar" type="bool">false</property>
+  </ignition-gui>
+</plugin>
+<plugin filename="GridConfig" name="Grid config">
```
Can we document the child elements (e.g., `cell_count`, `vertical_cell_count`) somewhere more visible?

chapulina replied:
Good call, documented on the header.

jennuine commented on examples/config/grid_config.config:
```diff
@@ -0,0 +1,36 @@
+  <ignition-gui>
+    <property type="string" key="state">docked</property>
+  </ignition-gui>
+  <cell_count>3</cell_count>
```
How about renaming to `<horizontal_cell_count>` for clarity?

chapulina replied:
I added that as an alias, so that both `cell_count` and `horizontal_cell_count` are accepted, with `horizontal_cell_count` overriding in case both are set.

The reasons to keep `cell_count` are:

* That corresponds to the `ign-rendering` API call
* That's backwards compatible with `Grid3D`

jennuine reviewed (approved):
LGTM, I left a minor comment & suggestion but nonblocking

chapulina commented:
I'm working on addressing @jennuine 's comments, and I'll also take the opportunity to consolidate `GridConfig` and `Grid3d` once and for all. This should close https://github.com/ignitionrobotics/ign-gui/issues/183.

chapulina commented:
I updated this PR to incorporate `Grid3D`'s functionality, and removed the code for `Grid3D`. The functionality added mainly consists of managing multiple grids:

`ign gui -v 4 -c examples/config/grid_config.config`

![grids](https://user-images.githubusercontent.com/5751272/144976909-0b81e53e-1b8c-4cc5-83ff-10c9f565355e.gif)

I also deleted the original `Grid3D` and now `GridConfig` is installed with both names.

@jennuine , I made quite a lot of changes from the original PR, mind taking another look? Thanks!

chapulina commented:
While looking at the `Grid3D` code, I noticed a lot of it was still commented out because we never completed the migration from `QWidget` to `QtQuick`. Here's a gif [with the original `Grid3D`](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-gui/pull-requests/62/page/1). I miss having multiple views.

![511624492-gridpr](https://user-images.githubusercontent.com/5751272/144977732-7356d686-22bb-44a1-a1d6-978f78e62468.gif)

So I'll just note here that the functionality that I migrated to `GridConfig` is just what was working on `ign-gui6`, not all the functionality from `ign-gui0`. Specifically, the addition and removal of grids from the plugin can come at a later time.
